### DISCUSSION TITLE
Meeting GBFS and GTFS-RT needs

### DISCUSSION THREAD

**Discussion by markstos** [valhalla/valhalla#3755] [General]
I'm curious if there are people that are using Valhalla for some of their routing needs, and a second service that handles either GBFS or GTFS-RT feeds. If so, which other service are you using?

Valhalla looks like it covers most of what we use OpenTripPlanner for, but if we were considering a switch, we'd need a solution for these as well. Thanks!

**Comment by kevinkreiser** [comment]
GBFS is definitely something we should think about integrating at some point. Currently we use data that is already present within OSM but allowing the import of other feeds would certainly make the algorithm that handles the route finding more useful assuming there is decent coverage from 3rd party feeds.

Regarding GTFS-RT it would be cool to support that but will be, potentially, a lot more work to support fully. I could see supporting it in phases, maybe something like supporting closures at first, then adding support for delays and maybe forgoing entirely support for real time positioning of individual vehicles (could be useful but not immediately obvious as of this moment)

Long story short these are interesting and would be welcome additions. If you or someone you know is interested in contributing I'd be happy to help. We just had a very successful google summer of code contributor get a good chunk of native GTFS support into valhalla so we'd love to support more such efforts. Let me know!

**Reply by markstos** [reply]
Thanks for the update. My company may be more interesting in donating funds rather than time in the future. We need to do some planning internally about our post-OTP-1.x future. I'm not sure how OTP 2.x memory use would compare to OTP 1.0, but it seems that Valhalla's memory use is dramatically less than OpenTripPlanner! We are already starting to deploy Valhalla for some it's unique features, and it would be nice to not be hosting coverage for both OTP 2 and Valhalla for the same areas.

**Reply by kevinkreiser** [reply]
great to hear you're enjoying the software! if you're looking to fund the development of specific features i think @nilsnolde might be able to help you.

**Reply by nilsnolde** [reply]
Thanks @kevinkreiser :) Indeed, we'd be happy to support the implementation, please feel free to contact us on enquiry@gis-ops.com anytime. 

I've been working with OTP over the past months a bit more as well and agree, the RAM usage when building can get close to OSRM's while the actual routing is not really that fast (other profiles than transit, i.e. bike/car/ped, for v1.5; think it's also AStar, but no fancy pre-processing that I could see). 

Anyways, would be great to have better support for multi-modality in Valhalla for regular users, not necessarily enabling full-blown transit scenario planning like OTP does.

**Reply by markstos** [reply]
Thanks. I'll move the conversation along at RideAmigos and see where it goes.
